drake_ros/drake_ros/systems/movable_joints.py

```diff
@@ -0,0 +1,144 @@
+            return
+
+        expected_frame = joint.child_body().body_frame().name()
+
+        if expected_frame != feedback.header.frame_id:
+            # TODO(sloretz) fix tf2_geometry_msgs_py :(
+            # transformed_point = self._tf_buffer.transform(point_stamped, self._frame_id)
+            print("TODO accept feedback in different frame")
+            return
+
+        qw = feedback.pose.orientation.w
+        qx = feedback.pose.orientation.x
+        qy = feedback.pose.orientation.y
+        qz = feedback.pose.orientation.z
+
+        new_orientation = Quaternion(qw, qx, qy, qz)
+        prev_orientation = self._joint_prev_orientation[joint.name()]
+        orientation_diff = prev_orientation.inverse().multiply(new_orientation)
+        diff_aa = AngleAxis(orientation_diff)
+
+        joint_axis = self._joint_axis_in_child[joint.name()]
+        dot = numpy.dot(joint_axis, diff_aa.axis())
+        if dot > 0.999:
+            angle_inc = diff_aa.angle()
+        elif dot < -0.999:
+            angle_inc = -1 * diff_aa.angle()
+        else:
+            angle_inc = 0.
+
+        angle = self._joint_states[self._joint_indexes[joint.name()]] + angle_inc
+
+        if angle > joint.position_upper_limit():
+            angle = joint.position_upper_limit()
+        elif angle < joint.position_lower_limit():
+            angle = joint.position_lower_limit()
+
+        self._joint_states[self._joint_indexes[joint.name()]] = angle
+        self._joint_prev_orientation[joint.name()] = new_orientation
+
+    def _do_get_joint_states(self, context, data):
+        data.SetFromVector(self._joint_states)
+
+    def _make_revolute_marker(self, revolute_joint: RevoluteJoint_):
+        int_marker = InteractiveMarker()
+        int_marker.header.frame_id = revolute_joint.child_body().body_frame().name()
+        int_marker.name = revolute_joint.name()
+        int_marker.scale = 0.3
+
+        int_marker.pose.position.x = 0.
+        int_marker.pose.position.y = 0.
+        int_marker.pose.position.z = 0.
+        int_marker.pose.orientation.w = 1.
+        int_marker.pose.orientation.x = 0.
+        int_marker.pose.orientation.y = 0.
+        int_marker.pose.orientation.z = 0.
+
+        # Drake revolute axis is in frame F on parent
+        axis_hat = revolute_joint.revolute_axis()
+        self._joint_axis_in_child[revolute_joint.name()] = axis_hat
+
+        # What rotation would get the parent X axis to align with the joint axis?
+        # https://math.stackexchange.com/q/476311
+        x_axis = (1, 0, 0)
+        v = numpy.cross(x_axis, axis_hat)
+        c = numpy.dot(x_axis, axis_hat)
+        v_sub_x = numpy.array((
+            (0, -v[2], v[1]),
+            (v[2], 0, -v[0]),
+            (-v[1], v[0], 0)), dtype=numpy.float64)
+        v_sub_x_squared = numpy.dot(v_sub_x, v_sub_x)
+        rotation_matrix = numpy.eye(3) + v_sub_x + v_sub_x_squared * (1.0 / (1.0 + c))
```

Comment:
If `axis_hat` and `x_axis` are close to being anti-parallel, `c` will be close to -1.0 and cause a division by zero. I found a function in `pydrake.math` to compute this matrix.
```suggestion
        # What rotation would get the parent X axis to align with the joint axis?
        rotation_matrix = ComputeBasisFromAxis(0, axis_hat)
```
Feel free to ignore if this code is just for the iiwa.